src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -515,6 +515,18 @@ public ToolTipDescription ToolTipDescriptionFor(int index)
          };
       }
 
+      private void setUnitAndDimension(ColumnMappingDTO model)
+      {
+         var supportedDimensions = _columnInfos.First(x => x.DisplayName == model.MappingName).SupportedDimensions;
+         var unit = _format.ExtractUnitDescriptions(model.ExcelColumn, supportedDimensions);
+         if (unit.SelectedUnit != UnitDescription.InvalidUnit)
+         {
+            var mappingDataFormatParameter = (model.Source as MappingDataFormatParameter);
```

Comment:
are we sure this is a MappingDataFormatParameter. If yes, explain why as comment
I am seeing this type of casting with as also a lot in the code. This is very unusual to always know what to cast to and I believe there is a need for refactoring here as well

Reply:
(but not for this PR...it's a general observation)

Reply:
I think you are right. We definitely need a refactoring. The thing is DataFormatParameters have a lot of common use but some special features are concrete for mappings (time, measurement and error). I would say there could be a refactoring missing but I would like to postpone it for now.